cmp/cmpopts/struct_filter.go

```diff
@@ -170,7 +170,7 @@ func canonicalName(t reflect.Type, sel string) ([]string, error) {
 	if !ok {
 		return []string{name}, fmt.Errorf("does not exist")
 	}
-	var ss []string
+	ss := make([]string,0,len(sf.Index))
```

Comment:
I'm not fond of up-front slice allocation unless the code has shown a performance benefit from doing so.
None of these slice usages are in the hot-path. Thus, the readability loss is not worth the supposed performance gains.